tests/lib/formatters/visualstudio.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * @fileoverview Tests for VisualStudio format.
+ * @author Ronald Pijnacker
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var assert = require("chai").assert,
+    formatter = require("../../../lib/formatters/visualstudio");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+describe("formatter:visualstudio", function() {
+    describe("when passed no messages", function() {
+        var code = [{
+            filePath: "foo.js",
+            messages: []
+        }];
+
+        it("should return nothing", function() {
+            var result = formatter(code);
+            assert.equal(result, "");
+        });
+    });
+
+    describe("when passed a single message", function() {
+        var code = [{
+            filePath: "foo.js",
+            messages: [{
+                message: "Unexpected foo.",
+                severity: 2,
+                line: 5,
+                column: 10,
+                ruleId: "foo"
+            }]
+        }];
+
+        it("should return a string in the format filename(x,y):\t[Error]   z for errors", function() {
+            var result = formatter(code);
+            assert.equal(result, "foo.js(5,10):\t[Error]   Unexpected foo.\t(foo)\n\n1 problem");
```

Comment:
As far as I can tell, the output is just dumped into VisualStudio's output window, and it seems to only parse the filename part up to the colon. However, since there is a spec, we might as well follow it. I updated the code to adhere to it.

The "1 problem" at the end is ok (and nicely informative).